File: lesunny/029_콜라츠추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        long cal = num;
        
        while(cal != 1){
            answer++;
            if(cal%2 == 0){
                cal /= 2;
            }else{
                cal = cal*3+1;
            }
            
            if(answer > 500){
                answer = -1;
                break;
            }
        }
        return answer;
    }
}

File: JIWON/029 콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num_1) {
    int answer = 0;
    long num = num_1;
    
    while(num != 1){
        answer++;
        
        if(answer > 500){
            answer = -1;
            break;
        }
        
        if(num % 2 == 0){
            num = num / 2;
        } else{
            num = 3*num + 1;
        }
    }
    
    return answer;
}
